### PR TITLE
Extend user documentation with inter-workload dependencies

### PR DESCRIPTION
**Pull request by inf17101**
Issues: #48, #211 

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: doc/docs/reference/inter-workload-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+| Dependency type |  AddCondition         | meaning                                       |
+| --------------- | --------------------- | --------------------------------------------- |
+| running         | ADD_COND_RUNNING      | the dependency must be operational            |
+| succeeded       | ADD_COND_SUCCEEDED    | the dependency must be exited successfully    |
+| failed          | ADD_COND_FAILED       | the dependency must be failed                 |
+
+A user defines one or multiple dependencies for an workload by configuring the `AddCondition` for each dependency within the field `dependencies`:
+
+```yaml
+logger:
```
Comment (windsource) [inline doc/docs/reference/inter-workload-dependencies.md:28]: I would use the format of an Anakios manifest here, like

```yaml
apiVersion: v1
workloads:
  logger:
    ...
```

to get a better impression of the context.

Comment (inf17101) [reply doc/docs/reference/inter-workload-dependencies.md]: I have updated it.

**Comment by inf17101** [comment]
I have changed the writing style to a more technical writing style.

File: doc/docs/reference/inter-workload-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+## Explicit inter-workload dependencies
+
+Ankaios supports the following dependency types:
+
+| Dependency type |  AddCondition         | meaning                                       |
+| --------------- | --------------------- | --------------------------------------------- |
+| running         | ADD_COND_RUNNING      | The dependency must be operational.           |
+| succeeded       | ADD_COND_SUCCEEDED    | The dependency must exit successfully.        |
+| failed          | ADD_COND_FAILED       | The dependency must fail.                     |
```
Comment (windsource) [inline doc/docs/reference/inter-workload-dependencies.md:23]:
```suggestion
| Dependency type |  AddCondition         | Description                                       |
| --------------- | --------------------- | --------------------------------------------- |
| running         | ADD_COND_RUNNING      | The dependency must be operational.           |
| succeeded       | ADD_COND_SUCCEEDED    | The dependency must be successfully exited.        |
| failed          | ADD_COND_FAILED       | The dependency must exit with a non-zero return code.                     |
```

File: doc/docs/reference/inter-workload-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    agent: agent_B
+    runtimeConfig: |
+      image: alpine:latest
+      commandOptions: [ "--entrypoint", "/bin/sleep" ]
+      commandArgs: [ "2" ]
+  error_handler:
+    runtime: podman
+    agent: agent_A
+    dependencies:
+      storage_provider: ADD_COND_FAILED # (4)
```
Comment (windsource) [inline doc/docs/reference/inter-workload-dependencies.md:98]:
```suggestion
      storage_provider: ADD_COND_RUNNING # (1)!
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sleep" ]
      commandArgs: [ "3" ]
  storage_provider:
    runtime: podman
    agent: agent_B
    dependencies:
      init_storage: ADD_COND_SUCCEEDED # (2)!
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sh" ]
      commandArgs: [ "-c", "sleep 5; exit 1" ]
  init_storage: # (3)!
    runtime: podman
    agent: agent_B
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sleep" ]
      commandArgs: [ "2" ]
  error_handler:
    runtime: podman
    agent: agent_A
    dependencies:
      storage_provider: ADD_COND_FAILED # (4)!
```

Without this change the block cannot be copied. The `#`  needs to excluded (see https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#stripping-comments)

**Review by windsource — CHANGES_REQUESTED** [review]
Minor changes required.

**Comment by inf17101** [comment]
> Minor changes required.

Changes have been adopted.

File: doc/docs/reference/inter-workload-dependencies.md (hunk 3 above)
Comment (inf17101) [reply doc/docs/reference/inter-workload-dependencies.md]: It looks better without the comments. The feature is very helpful. Thanks. However, I think it is still marked as experimental in the latest doc, but was added in version 8.5.0 already (and in the release notes of 8.5.0 it was not marked as experimental and sounds like a real feature). I guess its fine.

Comment (windsource) [reply doc/docs/reference/inter-workload-dependencies.md]: We had this problem before and Ankaios will not accept YAML with a trailing `#`.

**Review by windsource — APPROVED** [review]
Ok now.

**Pull request merged**